vertx-web/src/main/java/io/vertx/ext/web/handler/impl/CorsHandlerImpl.java

```diff
@@ -121,9 +123,15 @@ public void handle(RoutingContext context) {
     HttpServerResponse response = context.response();
     String origin = context.request().headers().get(ORIGIN);
```

Comment:
Here's the first place you read a request header value in order to modify the response depending on the result, so you should set the `Vary` header right here:
```suggestion
    response.putHeader(VARY, ORIGIN);
    String origin = context.request().headers().get(ORIGIN);
```

(of course you could also extract those two lines to a separate method `readRequestHeaderForVary` - I don't know what the preferred style in this project is)

Reply:
Actually it could be

```suggestion
    response.putHeader(VARY, ORIGIN);
    String origin = request.headers().get(ORIGIN);
```